azure-maven-plugin-lib/src/main/java/com/microsoft/azure/maven/AbstractAzureMojo.java

```diff
@@ -517,6 +519,26 @@ public void execute() throws MojoExecutionException {
         }
     }
 
+    private static void initMavenSettingsProxy(MavenExecutionRequest request) {
+        if (request != null) {
+            final List<Proxy> mavenProxies = request.getProxies();
+            if (CollectionUtils.isNotEmpty(mavenProxies)) {
+                final Proxy mavenProxy = mavenProxies.stream().filter(
+                    proxy -> proxy.isActive() && proxy.getPort() > 0 && StringUtils.isNotBlank(proxy.getHost())).findFirst().orElse(null);
+                if (mavenProxy != null) {
+                    final ProxyManager.ProxyInfo mavenProxyInfo = ProxyManager.ProxyInfo.builder()
+                        .source("maven")
+                        .host(mavenProxy.getHost())
+                        .port(mavenProxy.getPort())
+                        .username(mavenProxy.getUsername())
+                        .password(mavenProxy.getPassword())
+                        .build();
+                    ProxyManager.getInstance().setActiveProxy(mavenProxyInfo);
```

Comment:
It seems that we need to call `ProxyManager.getInstance().applyProxy();` after set Active proxy, so shall we invoke apply method in `setActiveProxy`?

Reply:
Or shall we set active proxy to `AzureConfiguration` directly and then call `ProxyManager.getInstance().applyProxy();`

As its a bit confused that we set proxy to `ProxyManager` but get proxy configuration from Azure Configuration in `TokenCredentialManager` and `AzureCliUtils`

Reply:
updated